src/utils.ts

```diff
@@ -81,46 +82,51 @@ export const createSmartContractCallAddBatchParams = async (params: {
   result: string,
   hashParams: string
 }> => {
-  const {
-    donationsWithShare,
-    givRelayerAddress,
-    network
-  } = params;
-  if (donationsWithShare.length === 0) {
-    throw new Error('There is no eligible donations in this time range')
+  try {
+    const {
+      donationsWithShare,
+      givRelayerAddress,
+      network
+    } = params;
+    if (donationsWithShare.length === 0) {
+      throw new Error('There is no eligible donations in this time range')
 
-  }
-  const partNumbers = donationsWithShare.length / maxAddressesPerFunctionCall
-  const hashParams: any = {
-    ipfsLink: '',
-  }
-  console.log('createSmartContractCallAddBatchParams', {
-    givRelayerAddress,
-    network
-  })
-  let nonce = await getLastNonceForWalletAddress(givRelayerAddress, network)
-  const rawDatasForHash = []
-  for (let i = 0; i < partNumbers; i++) {
-    const smartContractBatchData = getSmartContractAddBatchesHash(
-      {
-        donationsWithShare: donationsWithShare.slice(i * maxAddressesPerFunctionCall, (i + 1) * maxAddressesPerFunctionCall),
-        nonce
+    }
+    const partNumbers = donationsWithShare.length / maxAddressesPerFunctionCall
```

Comment:
lets ensure maxAddressesPerFunctionCall is not 0

Reply:
We always call the function like this

https://github.com/Giveth/givback-calculation/blob/staging/src/index.ts#L211-L240


So this line prevents from being `0` or even `NaN`
 Number(maxAddressesPerFunctionCall) || 200

Reply:
You are right, it's better to add validation inside the function